src/klein/test/test_resource.py

```diff
@@ -115,16 +116,18 @@ def getWrittenData():
     return request
 
 
-def _render(resource, request, notifyFinish=True):
+def _render(
+    resource: KleinResource, request: IRequest, notifyFinish: bool = True
+) -> Deferred:
```

Comment:
Can we depend on `Deferred` being generic here, and give it a type?

Reply:
Needs #476